include/VirtualDesk.hpp

```diff
@@ -14,6 +14,7 @@
 #include <hyprland/src/helpers/memory/SharedPtr.hpp>
 
 typedef std::unordered_map<int, int>             WorkspaceMap;
+// map with CMonitor* -> hyprland workspace id
 typedef std::unordered_map<const CMonitor*, int> Layout;
```

Comment:
You could use this `Layout` map to retrieve the windows size for a virtual desk, thus avoiding most of the new changes.

Basically, iterating over all `vdesks` (from `VirtualDeskManager`'s `vdesksMap`), you could:

- iterate over all workspace ids for the current layout;
- For each workspace id `wid`, use `g_pCompositor->getWindowsOnWorkspace(wid)`

See also Hyprland original code at `src/debug/HyprCtl.cpp`, `getWorkspaceData` function.

Reply:
Yes you're right. 
In my thoughts the plugin had to do the requests over the socket and therefore I wanted to save on socket calls since hyprland is single threaded. 
Since we're a plugin and can get the internal state of hyprland cheaply this is definitely the better way to do this :)